### ISSUE TITLE
Blob X will not be processed because function Y cannot be found?

### ISSUE THREAD

**Issue by tlaukkanen** [Azure/azure-functions-host#10406]
#### Is your question related to a specific version? If so, please specify:

4.35.1.1

#### What language does your question apply to? (e.g. C#, JavaScript, Java, All) 

C# - .NET 8 isolated

#### Question

We updated our blob triggered function app to .NET 8 isolated but now it doesn't process any blobs. The error what we see in log stream is:
```
2024-08-19T13:45:36Z   [Verbose]   Poll for function 'BlobHandlerFunction' on queue 'azure-webjobs-...-blob-handler' with ClientRequestId '...' found 1 messages in 5 ms.
2024-08-19T13:45:36Z   [Verbose]   Blob 'my-folder/111_2024_08_19_10_22' will not be processed because function 'BlobHandlerFunction' cannot be found. Message Id: 'e0...75'
```
What could be the reason why it cannot find the function?

Function `BlobHandlerFunction` is shown in a function list in portal in Functions list with status "Enabled".

Function code as simplified:
```
        [Function("BlobHandlerFunction")]
        public async Task RunAsync(
            [BlobTrigger("%ContainerName%/%BlobFolder%/{name}",
              Source = BlobTriggerSource.EventGrid,
              Connection = "AzureWebJobsStorage")] Stream stream,
            string name,
            ILogger log,
            CancellationToken cancellationToken)
        {
```

Environment variables:
FUNCTIONS_EXTENSION_VERSION = ~4
FUNCTIONS_WORKER_RUNTIME = dotnet-isolated

**Comment by bhagyshricompany** [comment]
Hi @tlaukkanen  thanks for reporting please share the repro steps.

**Comment by tlaukkanen** [comment]
Ok, I think we found the root cause which was the Event Grid subscription's webhook URL not having function name properly set or then the Function itself should also have matching method name if it uses that for matching. So steps that we did to fix this was:
1. Event Grid webhook URL to have `Host.Functions.` prefix for function name, like:
`...?functionName=Host.Functions.BlobHandlerFunction&...`
2. Changed the function method name to match the function name:
```
[Function(nameof(BlobHandlerFunction))]
public async Task BlobHandlerFunction(
```

Closing the issue. Hopefully someone else finds this helpful for their `function cannot be found` errors 🙂
